tests/backwards_compatibility_tests/test_general_compatibility.py

```diff
@@ -138,36 +138,35 @@ def prepare(self):
-                    if index.get("type") is not None and index.get('type') == 'structured':
-                        self.client.index(index_name=index['indexName']).add_documents(documents=self.docs)
-                    else:
-                        self.client.index(index_name=index['indexName']).add_documents(documents=self.docs,
-                                                                                       mappings=self.mappings,
-                                                                                       tensor_fields=self.tensor_fields)
-            except Exception as e:
-                self.logger.error(f"Exception occurred while adding documents {e}")
+        try:
+            self.logger.debug(f'Feeding documents to {self.indexes_to_test_on}')
+            for index in self.indexes_to_test_on:
+                if index.get("type") is not None and index.get('type') == 'structured':
+                    self.client.index(index_name=index['indexName']).add_documents(documents=self.docs)
+                else:
+                    self.client.index(index_name=index['indexName']).add_documents(documents=self.docs,
+                                                                                   mappings=self.mappings,
+                                                                                   tensor_fields=self.tensor_fields)
+        except Exception as e:
+            self.logger.error(f"Exception occurred while adding documents {e}")
+            raise e
 
         all_results = {}
         # Loop through queries, search methods, and result keys to populate unstructured_results
-        with self.subTest("Storing search results to be compared later"):
-            for index in self.indexes_to_test_on:
-                index_name = index['indexName']
-                all_results[index_name] = {}
-
-                # For each index, store results for different search methods
-                for query, search_method, result_key in zip(self.queries, self.search_methods, self.result_keys):
-                    if index.get("type") is not None and index.get("type") == 'structured':
-                        if search_method == 'HYBRID':
-                            result = self.client.index(index_name).search(q=query, search_method=search_method, hybrid_parameters=self.hybrid_search_params)
-                        else:
-                            result = self.client.index(index_name).search(q=query, search_method=search_method, searchable_attributes=self.searchable_attributes[search_method])
+        for index in self.indexes_to_test_on:
+            index_name = index['indexName']
+            all_results[index_name] = {}
+
+            # For each index, store results for different search methods
+            for query, search_method, result_key in zip(self.queries, self.search_methods, self.result_keys):
+                if index.get("type") is not None and index.get("type") == 'structured':
```

Comment:
semi-structured index also support searchable attributes.

Reply:
Will cover this in the next PR